lib/ansible/modules/cloud/amazon/ec2_vpc_egress_igw.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ec2_vpc_egress_igw
+short_description: Manage an AWS VPC Egress Only Internet gateway
+description:
+    - Manage an AWS VPC Egress Only Internet gateway
+version_added: "2.5"
+author: Daniel Shepherd (@shepdelacreme)
+options:
+  vpc_id:
+    description:
+      - The VPC ID for the VPC that this Egress Only Internet Gateway should be attached.
+    required: true
+  state:
+    description:
+      - Create or delete the EIGW
+    default: present
+    choices: [ 'present', 'absent' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Ensure that the VPC has an Internet Gateway.
+# The Internet Gateway ID is can be accessed via {{eigw.gateway_id}} for use in setting up NATs etc.
+ec2_vpc_egress_igw:
+  vpc_id: vpc-abcdefgh
+  state: present
+register: eigw
+
+'''
+
+RETURN = '''
+gateway_id:
+    description: The ID of the Egress Only Internet Gateway or Null.
+    returned: always
+    type: string
+    sample: eigw-0e00cf111ba5bc11e
+vpc_id:
+    description: The ID of the VPC to attach or detach gateway from.
+    returned: always
+    type: string
+    sample: vpc-012345678
+'''
+
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    camel_dict_to_snake_dict
+)
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be picked up by HAS_BOTO3 in AnsibleAWSModule
+
+
+def delete_eigw(module, conn, eigw_id):
+    """
+    Delete EIGW.
+
+    module     : AnsibleModule object
+    conn       : boto3 client connection object
+    eigw_id    : ID of the EIGW to delete
+    """
+    changed = False
+
+    try:
+        response = conn.delete_egress_only_internet_gateway(DryRun=module.check_mode, EgressOnlyInternetGatewayId=eigw_id)
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        # When boto3 method is run with DryRun=True it returns an error on success
+        # We need to catch the error and return something valid
+        if "DryRunOperation" in e.message:
```

Comment:
e.message unfortunately doesn't exist at all on Python3. What you had before looked right. You could either break up 
```
try:
    boto_call
except ClientError as e:
    < things with e.response and fail_json_aws >
except BotoCoreError as e:
    < just fail_json_aws >
```
Or you could wrap the e.response in a `try`/`except AttributError`

Reply:
I'm working on a PR to update CI for these tests right now. :)

Reply:
ah ok I didn't test in Py3. I think I will split the excepts instead of adding another try/except. Thank you!